migrations/update_filtered_positives.py

```diff
@@ -39,41 +40,56 @@ def run(settings_module: str = "") -> None:
-                version = filtered_positive_identifier.version
-                update_timestamp = datetime.now()
-                logger.info("Updating filtered positives...")
-                update_filtered_positive_fields(
-                    filtered_positive_identifier,
-                    positive_pending_samples,
-                    version,
-                    update_timestamp,
-                )
-                logger.info("Updated filtered positives")
-
-                logger.info("Updating Mongo...")
-                mongo_updated = update_mongo_filtered_positive_fields(
-                    config, positive_pending_samples, version, update_timestamp
-                )
-                logger.info("Finished updating Mongo")
-
-                if mongo_updated:
-                    logger.info("Updating MLWH...")
-                    mlwh_updated = update_mlwh_filtered_positive_fields(config, positive_pending_samples)
-                    logger.info("Finished updating MLWH")
-
-                    if mlwh_updated:
-                        logger.info("Updating DART...")
-                        dart_updated = update_dart_fields(config, positive_pending_samples)
-                        logger.info("Finished updating DART")
+
+                # Filter out cherrypicked samples
+                logger.info("Filtering out cherrypicked samples...")
+                non_cp_pos_pending_samples = remove_cherrypicked_samples(config, positive_pending_samples)
+
+                if num_non_cp_pos_pending_samples := len(non_cp_pos_pending_samples):
+                    logger.info(
+                        f"{num_non_cp_pos_pending_samples} non-cherrypicked positive samples in pending plates found"
+                    )
+                    filtered_positive_identifier = current_filtered_positive_identifier()
+                    version = filtered_positive_identifier.version
+                    update_timestamp = datetime.now()
+                    logger.info("Updating filtered positives...")
+                    update_filtered_positive_fields(
+                        filtered_positive_identifier,
+                        non_cp_pos_pending_samples,
+                        version,
+                        update_timestamp,
+                    )
+                    logger.info("Updated filtered positives")
+
+                    logger.info("Updating Mongo...")
+                    mongo_updated = update_mongo_filtered_positive_fields(
+                        config, non_cp_pos_pending_samples, version, update_timestamp
+                    )
+                    logger.info("Finished updating Mongo")
+
+                    if mongo_updated:
+                        logger.info("Updating MLWH...")
+                        mlwh_updated = update_mlwh_filtered_positive_fields(config, non_cp_pos_pending_samples)
+                        logger.info("Finished updating MLWH")
+
+                        if mlwh_updated:
```

Comment:
Should we add a check or setting in the event that we don't go with Beckman but still want to run this migration?

Reply:
I think this is something we should be able to do. We'd just not get pending plate barcodes from DART, instead getting all RESULT=Positive samples from mongo, and not update DART at the end. Do you think it worth it? It'd be a little extra work, but not too much

Reply:
Tough call. They could delay the decision on Beckman till next year sometime but we might want to run this sooner? We have some time while we wait on unified_warehouse etc. so we can see if there is any news tomorrow morning?

Reply:
I think it worthwhile as it allows more separation of the Beckman workflow and the fit-to-pick rules. My intention would be to implement it opposite to the normal crawler, i.e. attempt to query DART unless told not to

Reply:
Done